Realm/RLMObjectBase.mm

```diff
@@ -133,24 +106,48 @@ + (RLMObjectSchema *)sharedSchema {
     return RLMSchema.sharedSchema[self.className];
 }
 
-- (NSArray *)linkingObjectsOfClass:(NSString *)className forProperty:(NSString *)property {
-    if (!_realm) {
+void RLMObjectBaseSetRealm(__unsafe_unretained RLMObjectBase *object, __unsafe_unretained RLMRealm *realm) {
```

Comment:
I gave the same feedback on the original pr so I agree.

Also interspersing c code in an objc implementation is pretty annoying. Will move these out.